src/GraphQL/DI/IConfigureExecution.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+
+namespace GraphQL.DI
+{
+    /// <summary>
+    /// Allows configuration of execution options immediately prior to executing a document.
+    /// </summary>
+    public interface IConfigureExecution
+    {
+        /// <summary>
+        /// Configures execution options immediately prior to executing a document.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="ExecutionOptions.RequestServices"/> can be used to resolve other services from the dependency injection framework.
+        /// </remarks>
+        void Configure(ExecutionOptions executionOptions);
+    }
+
+    internal class ConfigureExecution : IConfigureExecution
+    {
+        private readonly Action<ExecutionOptions> _action;
+
+        public ConfigureExecution(Action<ExecutionOptions> action)
+        {
+            _action = action ?? throw new ArgumentNullException(nameof(action));
```

Comment:
I would not use any `throw new ArgumentNullException(nameof(XXX));` in new code. I think that the era of such design went into the past.

Reply:
I certainly would do so.  Why would you even say that?  Nullable reference types help, of course, but that doesn't mean it's an obsolete standard.  It is much better to have a proper argument null exception than to have a random null reference exception buried deeper down the stack trace.  If there's any new recommendation by Microsoft please forward me the documentation.

Reply:
I understand all you said. Yes, they only help, not 100% guard. Yes, there is no any new recommendation by Microsoft.

> It is much better to have a proper argument null exception than to have a random null reference exception buried deeper down the stack trace. 

My experience shows that in fact an exception is thrown almost immediately, there is practically no difference. But the code is clogged by all these `?? throw new ArgumentNullException()`. I do not insist to remove checks. Just a note from me. Personally I would prefer to use this https://github.com/dotnet/csharplang/issues/2145 when/if it will be ready.